src/components/connect.js

```diff
@@ -17,9 +17,36 @@ function getDisplayName(WrappedComponent) {
 // Helps track hot reloading.
 let nextVersion = 0
 
-export default function connect(mapPropsToRequestsToProps, options = {}) {
+function connectFactory(options = {}) {
+  function connectImpl(map) {
+    return connect(map, options)
+  }
+
+  connectImpl.withFetch = function (fetch) {
+    invariant(
+      typeof fetch === 'function',
+      '`withFetch` must be passed a function. Instead received a %s.',
+      typeof fetch
+    )
+
+    return connectFactory(assign({}, options, { fetch: fetch }))
+  }
+
+  connectImpl.withRef = function (withRef = true) {
+    return connectFactory(assign({}, options, { withRef: withRef }))
+  }
+
+  return connectImpl
+}
+
+export default connectFactory()
+
+function connect(mapPropsToRequestsToProps, options) {
   const finalMapPropsToRequestsToProps = mapPropsToRequestsToProps || defaultMapPropsToRequestsToProps
-  const { withRef = false } = options
+  const {
+    fetch = window.fetch,
+    withRef = false
```

Comment:
i think we can default this to `true` or probably even just remove it as an option, unless someone has a good reason you'd want to be able to not have a ref. See https://github.com/heroku/react-refetch/issues/60#issuecomment-187852641